src/en/charms-constraints.md

```diff
@@ -1,187 +1,233 @@
+Title: Using constraints
+TODO:  Important: include default resources requested for non-constrained machine
+       bug tracking: https://bugs.launchpad.net/juju/+bug/1768308
 
-# Constraints
+# Using constraints
 
-Constraints allow you to choose the hardware (or virtual hardware)
-to which your applications will be deployed, e.g. by specifying the amount of 
-RAM you want them to have. This is particularly useful for making sure that the
-application is deployed somewhere it can actually run efficiently, or that it is
-connected to the right network. Constraints may be set for models and
-applications, with application constraints taking precedence. Changes to
-constraints do not affect any units which have already been placed on machines.
+A *constraint* is a user-defined minimum hardware specification for a machine
+that is spawned by Juju. There are a total of nine constraints, with the most
+common ones being 'mem', 'cores', 'root-disk', and 'arch'. The definitive
+constraint resource is found on the
+[Reference: Juju constraints][reference-constraints] page.
 
-For more granularity, it is also possible to add a machine with specific 
-constraints (`juju add-machine`) and then specify that machine when deploying 
-applications ([see the documentation on `juju deploy`](./charms-deploying.html)).
+The idealized use case is that of stipulating a constraint when deploying an
+application and the backing cloud providing a machine with those exact
+resources. In the majority of cases, however, default constraints may have been
+set (at various levels) and the cloud is unable to supply those exact
+resources.
 
-## What constraints can be used?
+Several noteworthy constraint characteristics:
 
-There is a full list of the constraints used 
-[in the reference section](reference-constraints.html). Be aware that some of 
-these are specific to the type of cloud you are using. For example, one may
-understand an "instance-type" constraint, but another may not. 
+ - Some constraints are only supported by certain clouds.
+ - Changes to constraint defaults do not affect existing machines.
+ - Multiple constraints are logically AND'd (i.e. the machine must satisfy all
+   constraints).
 
-The most useful constraints for Juju in general are:
-  
-  - **mem** : This indicates the minimum number of megabytes of RAM that must 
-  be available to an application unit. An optional suffix of M/G/T/P indicates
-  the value is mega-/giga-/tera-/peta- bytes.
-
-  - **cores** :  How many cpu cores the host machine should have. This is a
-  crude indicator of system performance.
-    
-  - **spaces** : Target a particular network space, or avoid one. See 
-  [Network spaces][network-spaces].
-  
-  - **arch** : Short for 'architecture', indicates the processor type an
-  application must run on. One of amd64, arm, i386, arm64, or ppc64el.
-  
-With these you can make sure an application has the resources it needs to run 
-properly.
+## Clouds and constraints
```

Comment:
I don't think this should go in a separate section, this is common to ALL clouds/non-clouds and really, I think these things should be in the 'noteworthy' list.

Reply:
But the characteristics are short bullet points and after bringing down the 'idealised case' there are now three paragraphs. How do I combine these with the bullet points?